### ISSUE TITLE
schemaName support

### ISSUE THREAD

**Issue by dzuluaga** [balderdashy/sails-postgresql#207]
Specifying the schemaName in the model doesn't seem to store data into user table from another schema. It keeps storing it to public schema. For instance, the following lines will create a record in public.user table instead of testmyapischema. Am I missing anything?

Controller:
```javascript
    User.create({name:'Diego'}).exec(function createCB(err, created){
      console.log('Created user with name ' + created.name);
    });
```
Model:
```javascript
module.exports = {
  connection : 'somePostgresqlServer',
  tableName : 'user',
  meta: {
       schemaName: 'testmyapischema'
    },

  attributes : {
    name : {
      type : 'string'
    },
    age : {
      type : 'integer'
    }
  }
};

```

**Comment by devinivy** [comment]
Where did you find this feature documented?

**Comment by dzuluaga** [comment]
It's not docummented as per [this thread](http://stackoverflow.com/questions/24380902/database-schema-with-sails-js-and-sails-postgresql) on SO. As per @sgress454 this should work. But in my experience, it still creates and reads the record from public schema not the schema specified on the model.

**Comment by dzuluaga** [comment]
@devinivy, @particlebanana do you happen to know any clue on this? I wonder if instead of the adapter, I can test it with another adapter such as sequelize.js through hooks https://www.npmjs.com/package/sails-hook-sequelize. However, I leveraging other ORMs may lose some benefits of the integration features with sails.js.

**Comment by devinivy** [comment]
It looks like this may have been dropped by mistake during the migration to use waterline-sequel for query-building: https://github.com/balderdashy/sails-postgresql/commit/3b3d616652f2a5c4c25546bc7afbe6dd400d281b

Options for schema-name will need to be added to waterline-sequel to restore this feature.  It shouldn't be _terribly_ hard, but someone will have to do it.  You might also check-out the new [https://github.com/waterlinejs/postgresql-adapter](postgresql-adapter) that's being worked-on.

**Comment by dzuluaga** [comment]
Thanks for your answer @devinivy. I tried postgresql-adapter, and I'm experiencing the same issue. It only works with a public schema. Supporting custom schemas is rather critical for my project. I wonder how this issue has been workaround by others. Thanks again!

**Comment by justmiles** [comment]
Ran into this issue too. Had to revert to 0.10.11 and add the schema inside the `tableName` attribute.
```
  tableName: 'schema.table'
```

**Comment by sailsbot** [comment]
Thanks for posting, @dzuluaga.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by sgress454** [comment]
Thanks @devinivy, looking at this today.

**Comment by sgress454** [comment]
Ok, thanks for everyone's patience.  We pushed this out today as v0.11.0. Suffice it to say, it was a lot more involved than we originally thought.

The solution we came up with fleshes out support for the `meta.schemaName` property, including automatically creating schemas for models when `migrate` is set to `alter` or `create`.  The biggest caveat is that _we do not currently support models with the same `tableName` in different schemas_.  I realize that having duplicate tables is a common use-case for Postgresql schemas.  Waterline is quite simply not designed for this kind of thing--at the lower levels, in deals exclusively with table names and expects them to be unique.  If this is a showstopper for a significant number of developers, then it's something that may well be worth looking into, but for now you can at least set `meta.schemaName` on a per-model basis and it will work.

**Comment by Hardware-Hacks** [comment]
Was all looking great until the caveat :( 

I rename and then recreate my db schema a number of times a year. I end up with public being the current live schema and a nice naming convention for older versions which I'd like to use for development. The tables/constructs are named in the exact same way. It sounds as though the only way I can fix this is to rename my tables during the migration to work with this fix?

**Comment by sgress454** [comment]
@Hardware-Hacks Actually I think your use case might be fine, if I understand you correctly.  What you can't do is have two models in the same app, `Foo` and `Bar`, use the same table name but different schemas.  If all you're wanting to do is switch the schema that `Foo` points to conditionally (probably based on the environment), you _should_ be able to do that.  Try creating a `config/env/development/models.js` file and a `config/env/production/models.js` file, and setting `meta.schemaName` in each of those, then check what happens when you lift Sails in production (`sails lift --prod`) vs. development.

**Comment by Hardware-Hacks** [comment]
@sgress454 excellent thanks for the clarification I will test this out tonight and let you know the outcome, however the scenario you explain is the use case I have so all should be fine :+1:

**Comment by Hardware-Hacks** [comment]
Worked perfectly thanks

`   models: {
     connection: 'Postgresql',
    meta: {
      schemaName: 'r65'
    }
   }`

**Comment by Hardware-Hacks** [comment]
Out of interest.  Do you know if I can utilise similar code for my .query executions to point to a different schema?. I appreciate that this bypasses models to run raw queries against the db but would be nice to differentiate dev and prod ones.
